src/manybody_nequip.F

```diff
@@ -0,0 +1,517 @@
+                  CALL sort(sort_list(1, :), npairs, work_list)
+                  DO ipair = 1, npairs
+                     work_list(ipair) = sort_list(2, work_list(ipair))
+                  END DO
+                  sort_list(2, :) = work_list
+                  ! find number of unique elements of array index 1
+                  nunique = 1
+                  DO ipair = 1, npairs - 1
+                     IF (sort_list(1, ipair + 1) /= sort_list(1, ipair)) nunique = nunique + 1
+                  END DO
+                  ipair = 1
+                  junique = sort_list(1, ipair)
+                  ifirst = 1
+                  DO iunique = 1, nunique
+                     atom_a = junique
+                     IF (glob_loc_list_a(ifirst) > atom_a) CYCLE
+                     DO mpair = ifirst, SIZE(glob_loc_list_a)
+                        IF (glob_loc_list_a(mpair) == atom_a) EXIT
+                     END DO
+                     ifirst = mpair
+                     DO mpair = ifirst, SIZE(glob_loc_list_a)
+                        IF (glob_loc_list_a(mpair) /= atom_a) EXIT
+                     END DO
+                     ilast = mpair - 1
+                     nloc_size = 0
+                     IF (ifirst /= 0) nloc_size = ilast - ifirst + 1
+                     DO WHILE (ipair <= npairs)
+                        IF (sort_list(1, ipair) /= junique) EXIT
+                        atom_b = sort_list(2, ipair)
+                        rij(:) = r_last_update_pbc(atom_b)%r(:) - r_last_update_pbc(atom_a)%r(:) + cell_v
+                        drij = DOT_PRODUCT(rij, rij)
+                        ipair = ipair + 1
+                        IF (drij <= rab2_max) THEN
+                           nedges = nedges + 1
+                           edge_index(:, nedges) = [atom_a - 1, atom_b - 1]
+                           edge_cell_shifts(:, nedges) = REAL(cvi, kind=sp)
+                        END IF
+                     END DO
+                     ifirst = ilast + 1
+                     IF (ipair <= npairs) junique = sort_list(1, ipair)
+                  END DO
+                  DEALLOCATE (sort_list, work_list)
+               END IF
+            END DO
+         END DO Kind_Group_Loop_Nequip
+      END DO
+
+      ALLOCATE (edge_count(para_env%num_pe))
+      ALLOCATE (edge_count_cell(para_env%num_pe))
+      ALLOCATE (displ_cell(para_env%num_pe))
+      ALLOCATE (displ(para_env%num_pe))
+
+      CALL mp_allgather(nedges, edge_count, para_env%group)
+      nedges_tot = SUM(edge_count)
+
+      ALLOCATE (temp_edge_index(2, nedges))
+      temp_edge_index = edge_index(:, :nedges)
+      DEALLOCATE (edge_index)
+      ALLOCATE (temp_edge_cell_shifts(3, nedges))
+      temp_edge_cell_shifts = edge_cell_shifts(:, :nedges)
+      DEALLOCATE (edge_cell_shifts)
+
+      ALLOCATE (edge_index(2, nedges_tot))
+      ALLOCATE (edge_cell_shifts(3, nedges_tot))
+      ALLOCATE (t_edge_index(nedges_tot, 2))
+
+      edge_count_cell = edge_count*3
```

Comment:
This fails the build because of `-Werror=realloc-lhs`. You can fix this by explicitly assigning to the entire array:
```
edge_count_cell(:) = edge_count*3
```

Reply:
this passes now.